ETL support: Allow data to be sourced at MERGE time

eugeneniemand opened the pull request:
If sourcing large volume of records the statement can reach the EXEC() limits due to the generation of the VALUES statement. Using a the table instead of VALUES is useful for large tables as it can leverage indexes and the merge statement can be reused as part of an ETL. We have used this pattern on Data Warehouse Projects

eugeneniemand committed:
Added option to use full source table

If sourcing large volume of records the statement can reach the EXEC() limits due to the generation of the VALUES statement. Using a the table instead of VALUES is useful for large tables as it can leverage indexes and the merge statement can be reused as part of an ETL. We have used this pattern on Data Warehouse Projects

dnlnln commented:
I can definitely see the use case for the addition of the `@table_not_values` option, although I'm wondering if the parameter name may need a bit of finessing. Given that `sp_generate_merge` is generally used to track changes to static data in source control and used within automated release tools (e.g. SSDT, FlyWay, SQL Change Automation etc), not having the data in the generated statement is a bit of a departure from the typical set of use cases.

In that spirit, the name `@table_not_values` would suggest to me that running the tool with this option set to `1` would still result in the full table data being scripted, except that it would do so via `INSERT`s into a temporary table first. Whereas this PR results in the data being transferred directly from the source table, without any data actually being scripted.

Therefore, perhaps an alternate name for this option would make sense, e.g. `@include_values` (default: 1), or `@table_as_source` (default: 0). Thanks for your submission and it would be great to hear your thoughts.

Note that this PR partially duplicates the work in PR #34. Once this and a couple of related PRs are merged, #34 will be closed.

dnlnln committed:
Merge branch 'master' into patch-1

# Conflicts:
#	master.dbo.sp_generate_merge.sql

dnlnln committed:
Rename @table_not_values (0) param to @include_values (1)
Per https://github.com/readyroll/generate-sql-merge/pull/37#issuecomment-449595877

dnlnln commented:
To select directly from the source table when the `MERGE` statement is executed (rather than including static within a `VALUES` clause as the source), set the newly-added `@include_values` to `0`:

```
EXEC sp_generate_merge 'AddressType', @schema = 'Person', @include_values = 0, @target_table = 'Person_Copy'
```
